### PR TITLE
conditional_simple programs with decision points_hard_q1

### PR DESCRIPTION
**Pull request by SamamthaH**

### CODE REVIEW

File: src/conditionals/simple_programs_with_decision_points/hard/q1/AbsolutelySureTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("Enter a number: "),
+                new NewLine(),
+                new IntegerLiteral(5, "returnedValue")
+        };
+    }
+
+    public void runMain() {
+        provideInput("5");
```
Comment (SanaBob) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/AbsolutelySureTest.java:20]: put 5 in it's own variable at the top of the class and input it here

File: src/conditionals/simple_programs_with_decision_points/hard/q1/AbsolutelySureTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AbsolutelySureTest extends BaseTest {
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("Enter a number: "),
+                new NewLine(),
+                new IntegerLiteral(5, "returnedValue")
```
Comment (SanaBob) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/AbsolutelySureTest.java:15]: same here you can assign 5 to its own variable and input it here which makes it less hardcoded

File: src/conditionals/simple_programs_with_decision_points/hard/q1/AbsolutelySureTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        provideInput(String.valueOf(input));
+        AbsolutelySure.main(new String[]{});
+    }
+
+    @Test
+    public void negativeValueTest() {
+        provideInput("-46");
+        AbsolutelySure.main(new String[]{});
+        String output = getOutput();
+        int contains = output.indexOf("-46");
```
Comment (SanaBob) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/AbsolutelySureTest.java:31]: you can put -46 in its own variable and then provideInpit(variable) and indexOf(variable) to look less hardcoded

File: src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest extends BaseTest {
+    //Parsons
```
Comment (Hedgemon4) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:11]: `\\ Parsons with distractors`

File: src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                new IntegerLiteral(input, "returnedValue")
+        };
+    }
+
+    public void runMain() {
+        provideInput(String.valueOf(input));
+        AbsolutelySure.main(new String[]{});
+    }
+
+    @Test
```
Comment (Hedgemon4) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:28]: You might want to consider using the inputTest beta for this question

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(Arguments.of(5,5), Arguments.of(-46, 46), Arguments.of(0, 0));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void testWithInput(int input, int output) {
+        String error = "The absolute value was calculated incorrectly. ";
+        if(input<0 )
+            error += "Negative values were no handled properly.";
```
Comment (Hedgemon4) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:41]: I presume this should say not instead of no

Comment (Hedgemon4) [reply src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:41]: I also feel that I don't think the message is fantastic if you deliver both, maybe you should have different messages depending on what happens

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java (hunk 6 above)
Comment (SamamthaH) [reply src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:41]: what do you mean? If it fails a test for a positive value it just says the base error message but then it adds the customized negative one if it doesn't convert the negative to a positive.

File: src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(Arguments.of(5,5), Arguments.of(-46, 46), Arguments.of(0, 0));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void testWithInput(int input, int output) {
+        String error = "The absolute value was calculated incorrectly. ";
+        if(input<0 )
```
Comment (Opeyem1a) [inline src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:40]: format file

Comment (Opeyem1a) [reply src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java:40]: If you're going to have custom error messages based on if it handles different cases well, hardcode that as an extra argument from the `inputProvider`. Whether or not you do that, they should be more descriptive if you're going through the trouble of making them distinct. 

i.e. "The absolute value of positive numbers is ..."

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**